anchore_engine/analyzers/syft.py

```diff
@@ -3,11 +3,82 @@
 import os
 import collections
 
-import anchore_engine.analyzers.utils
+from anchore_engine.analyzers.utils import dig, defaultdict_to_dict
 from anchore_engine.clients.syft_wrapper import run_syft
 
 
+def handle_java(findings, artifact):
+    """Java results handler for syft output.
+
+    Args:
+        findings (dict): nested dictionary representing a json structure.
+        artifact (json): datastructure presented by syft.
+
+    Returns:
+        dict: findings dictionary is populated with values from the provided
+              artifact.
+    """
+    
+    pkg_key = dig(artifact, 'metadata', 'virtualPath', default="N/A")
+
+    virtualElements = pkg_key.split(":")
+    if "." in virtualElements[-1]:
+        # there may be an extension in the virtual path, use it
+        java_ext = virtualElements[-1].split(".")[-1]
+    else:
+        # the last field is probably a package name, use the second to last virtual path element and extract the extension
+        java_ext = virtualElements[-2].split(".")[-1]
```

Comment:
most/all of this block would benefit from getting extracted into a separate helper. By having it here in the `handle_java` function, it makes it hard to unit test (there aren't any that I could see for this) and makes the function too large, reducing the readability

Reply:
we haven't been writing unit tests for the analyzer module work, but instead are using the functional test fixtures to cover the necessary cases (consistent with the outlined strategy for the analyzer module work overall and consistent with the other analyzer module PRs). 

Without comments this function is ~30 lines of code, which I feel is not too large.

Reply:
That sounds reasonable to me. I was interested in the small if/else blocks, but I'm fine if you feel extracting them would be overkill.